src/server/dflycmd.cc

```diff
@@ -379,6 +379,7 @@ void DflyCmd::TakeOver(CmdArgList args, ConnectionContext* cntx) {
 
   // TODO: We should cancel blocking commands before awaiting all
   // dispatches to finish.
+  sf_->CancelBlockingCommands();
   if (!sf_->AwaitDispatches(timeout_dur, [self = cntx->owner()](util::Connection* conn) {
         // The only command that is currently dispatching should be the takeover command -
         // so we wait until this is true.
```

Comment:
I have a question regarding the last PR and not to this one.
If we are calling AwaitDispatches and before you are blocking all new commands with setting TAKEN_OVER state, why do we need the TransactionGuard ?

Reply:
I think it's for expirations if that makes sense. Otherwise I agree we don't think we really need it.

Reply:
Ok if the TransactionGuard is not really needed, we can remove the logic of set no expiration from transaction guard and just write a code to do the no expiration logic. Any we dont need to do now this change, just add a TODO